pkg/binding/bindingpolicy-resolution-broker.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package binding
+
+import (
+	"github.com/kubestellar/kubestellar/api/control/v1alpha1"
+	"github.com/kubestellar/kubestellar/pkg/abstract"
+	"github.com/kubestellar/kubestellar/pkg/util"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sync"
+)
+
+// ResolutionBroker allows for the registration of callback functions that
+// are called when a resolution is updated.
+// It also allows for the retrieval of resolutions for a given bindingPolicyKey.
+type ResolutionBroker interface {
+	// RegisterCallback adds a new callback function that will be called,
+	// on a separate goroutine, when a resolution is updated. The callback
+	// function should accept a bindingPolicyKey.
+	// There is no deduplication of callbacks.
+	RegisterCallback(func(bindingPolicyKey string))
```

Comment:
FYI, lesson learned in the legacy system: because go `func` values are not comparable, it is impossible to later add an "Unregister" method. If, OTOH, the thing registered were an interface with a callback method, then "Unregister" would be possible. Not a blocker for this PR.